### DISCUSSION TITLE
how can access google Signin in Private window

### DISCUSSION THREAD

**Discussion by rejithkrea** [abacritt/angularx-social-login#459] [General]
Its annoying to each time enabling third party cookies , as well as say users to enable third party cookie, is any methods  access google sign in private window also, Medium and stackoverflow are doing something to login private window

**Comment by Navjot-Sharma** [comment]
Try adding ux_mode: 'redirect' in providers array in app module.
`{
            id: GoogleLoginProvider.PROVIDER_ID,
            provider: new GoogleLoginProvider(
              'xxxxxx',
              {scope: 'email', ux_mode: 'redirect'}
            )
          },`

It will return token in query parameters.
Note: Update redirect_uri in google developers console.

**Reply by aroshanzamir** [reply]
I'm using version **1.2.5**. in this version   `ux_mode `does not exist in `GoogleInitOptions `
`"@abacritt/angularx-social-login": "^1.2.5",`


```
export interface GoogleInitOptions {
   oneTapEnabled?: boolean;
   scopes?: string | string[];
   prompt_parent_id?: string;
   prompt?: '' | 'none' | 'consent' | 'select_account';
}
```

Where should I set `ux_mode `to `redirect`?
